### PR TITLE
#316 add SendMQ abi

### PR DESCRIPTION
**Pull request by hunshenshi**

### CODE REVIEW

File: pkg/types/wasm/context.go [hunk 1]
```diff
@@ -187,3 +188,24 @@ func MustRuntimeResourceFromContext(ctx context.Context) *mapx.Map[uint32, []byt
 	must.BeTrue(ok)
 	return v
 }
+
+func WithMQ(ctx context.Context, mq *MqttClient) context.Context {
```
Comment (Liuhaai) [inline pkg/types/wasm/context.go:192]: WithMQ -> WithMQTT

File: pkg/types/wasm/instance.go [hunk 2]
```diff
@@ -72,6 +72,7 @@ type ABI interface {
 	SetSQLDB(addr, size int32) int32
 	GetSQLDB(addr, size, vmAddrPtr, vmSizePtr int32) int32
 	GetEnv(kAddr, kSize, vmAddrPtr, vmSizePtr int32) int32
+	SendMQ(topicAddr, topicSize, msgAddr, msgSize int32) int32
```
Comment (Liuhaai) [inline pkg/types/wasm/instance.go:75]: same here

**Review by Liuhaai — APPROVED** [review]

File: pkg/enums/config_type.go [hunk 3]
```diff
@@ -8,5 +8,6 @@ const (
 	CONFIG_TYPE__PROJECT_SCHEMA
 	CONFIG_TYPE__INSTANCE_CACHE
 	CONFIG_TYPE__PROJECT_ENV
+	CONFIG_TYPE__PROJECT_MQ
```
Comment (saitofun) [inline pkg/enums/config_type.go:11]: move to bottom
```go
_ // deprecated CONFIG_TYPE__CHAIN_CLIENT
CONFIG_TYPE__PROJECT_MQ
```

File: pkg/modules/deploy/instance_context.go [hunk 4]
```diff
@@ -18,6 +18,7 @@ func WithInstanceRuntimeContext(parent context.Context) (context.Context, error)
 		types.WithLoggerContext(types.MustLoggerFromContext(parent)),
 		types.WithWasmDBExecutorContext(types.MustWasmDBExecutorFromContext(parent)),
 		types.WithRedisEndpointContext(types.MustRedisEndpointFromContext(parent)),
+		types.WithMqttBrokerContext(types.MustMqttBrokerFromContext(parent)),
```
Comment (saitofun) [inline pkg/modules/deploy/instance_context.go:21]: MQTT broker used by wasm should create from project config but not global context

**Review by saitofun — COMMENTED** [review]
I will review after #319 merged

File: pkg/modules/deploy/instance_context.go (hunk 4 above)
Comment (Liuhaai) [reply pkg/modules/deploy/instance_context.go:21]: The purpose of this pr is to add `ws_sendmq` abi, the change you mentioned could be done in following pr.

**Pull request merged**